src/main/java/org/commcare/suite/model/Entry.java

```diff
@@ -86,18 +87,31 @@ public Text getText() {
     }
 
     public Hashtable<String, DataInstance> getInstances() {
+        return getInstances(null);
+    }
+    /**
+     *
+     * @param limitingList a list of instance names to restrict the returning set to; null
+     *                     if no limiting is being used
+     * @return a hashtable representing the data instances that are in scope for this Entry,
+     * potentially limited by @limitingList
+     */
+    public Hashtable<String, DataInstance> getInstances(Set<String> limitingList) {
         Hashtable<String, DataInstance> copy = new Hashtable<>();
         for (Enumeration en = instances.keys(); en.hasMoreElements(); ) {
             String key = (String)en.nextElement();
 
-            //This is silly, all of these are externaldata instances. TODO: save their
+            //This is silly, all of these are external data instances. TODO: save their
             //construction details instead.
             DataInstance cur = instances.get(key);
-            if (cur instanceof ExternalDataInstance) {
-                //Copy the EDI so when it gets populated we don't keep it dependent on this object's lifecycle!!
-                copy.put(key, new ExternalDataInstance(((ExternalDataInstance)cur).getReference(), cur.getInstanceId()));
-            } else {
-                copy.put(key, cur);
+            if (limitingList == null || limitingList.contains(cur.getInstanceId())) {
+                // Make sure we either aren't using a limiting list, or the instanceid is in the list
+                if (cur instanceof ExternalDataInstance) {
+                    //Copy the EDI so when it gets populated we don't keep it dependent on this object's lifecycle!!
+                    copy.put(key, new ExternalDataInstance(((ExternalDataInstance)cur).getReference(), cur.getInstanceId()));
```

Comment:
I'm a bit confused by this statement. Is the "dependence" a memory leak (IE: The EDI ends up keeping a reference to this object)?

Reply:
That comment/code isn't from this PR at all; you can see it was the same a few lines above. All that's changed here is adding the `if` statement on line 107

Reply:
aha! missed that entirely, sorry.